index.js

```diff
@@ -175,6 +175,12 @@
     hasMethod('concat')
   );
 
+  //  Foldable :: TypeClass
+  var Foldable = $.TypeClass(
+    'sanctuary/Foldable',
+    hasMethod('reduce')
```

Comment:
We should return `true` for Array values explicitly. Even setting aside the fact that `Array#reduce` isn't defined in ancient environments, we should make it clear that there are two cases:

  - Array values, which we handle ourselves since `Array#reduce` is not :rainbow:-compliant; and
  - non-Array values with `reduce` methods.

Also, for lack of a better system I've been alphabetizing the type-class definitions. ;)